h3/h3.py

```diff
@@ -628,10 +628,10 @@ def hex_ranges(h3_address_list, ring_size):
             # divided by 6 ceiled.
             ring_index = int(
                 math.floor(
-                    (1 + math.sqrt(1 + 8 * math.ceil(j / 6.0))) / 2)) - 1
+                    (1 + math.sqrt(1 + 8 * math.max(math.ceil(j / 6.0 - 1), -0.1))) / 2))
```

Comment:
Sorry, that was poorly thought out. Revised, and tested against your tests:
```
        ring_index = 0
        ring_end = 0
        for j in range(0, range_size):
            if j > ring_end:
                ring_index = ring_index + 1
                ring_end = ring_end + 6 * ring_index
```

Reply:
No complex math at all, no C call, just incremental additions.

This actually brings the code in line with the native Python (though still not faster):
```
$ python -m timeit -s "from h3 import h3; ring = list(h3.k_ring('8928308280fffff', 20))" "h3.hex_ranges(ring, 20)"
1 loop, best of 5: 1.47 sec per loop
16:40 $ python -m timeit -s "from h3 import h3; ring = list(h3.k_ring('8928308280fffff', 20))" "{hex: h3.hex_range(hex, 20) for hex in ring}"
1 loop, best of 5: 1.38 sec per loop
```

Reply:
> This is a closed form solution that doesn't require iteration, it should quickly outpace that form, especially by being able to avoid the FFI call in-between. This version was written specifically because an approach like that was tried before this project was open sourced and was much slower.

You might be right with the FFI call, but as you can see we didn't actually need it (this is just the ring size, not the k-ring size, so it's just 6 * k). I don't understand the "avoiding iteration" argument here, because we're already iterating (though of course it would hold for computing k-ring size). In this case we do no math at all in the majority of iterations.

Anyway, I really think benchmarks should win performance arguments. I am perfectly happy to be proven wrong, but the benchmarks I've run locally so far show that this incremental version is 2x faster than the closed form version.